python_modules/dagster/dagster/core/definitions/output.py

```diff
@@ -257,6 +257,19 @@ def combine_with_inferred(self: TOut, inferred: InferredOutputProps) -> TOut:
             asset_partitions_def=self.asset_partitions_def,
         )
 
+    def with_asset_key(self, asset_key: AssetKey) -> "OutputDefinition":
```

Comment:
luckily a lot of this painful stuff won't be necessary in the future (when we move away from OutputDef.asset_key